todo_app/app.py

```diff
@@ -1,48 +1,82 @@
+from flask_login.mixins import AnonymousUserMixin
 from todo_app.data.MongoDbApi import MongoDbApi
 from todo_app.ViewModel import ViewModel
 from todo_app.data.Items import Items
 from flask import Flask, render_template, request, redirect, send_from_directory
 from todo_app.flask_config import Config
 import os
+from flask_login import LoginManager, login_required, current_user
+from todo_app.security.Auth import Auth
 
 
 def create_app(db_name=None):
     app = Flask(__name__)
     app.config.from_object(Config())
     MongoDbApi.init(db_name)
+    Auth.init()
 
     @app.route('/')
+    @login_required
     def index():  # pylint:disable=unused-variable
-        item_view_model = ViewModel(
-            Items.get_items(), Items.get_current_sort_order())
+        item_view_model = None
+        if app.config['LOGIN_DISABLED']:
+            item_view_model = ViewModel(
+                Items.get_items(), Items.get_current_sort_order(), True)
+        else:
+            item_view_model = ViewModel(Items.get_items(
+            ), Items.get_current_sort_order(), current_user.has_write_permission())
         return render_template('index.html', view_model=item_view_model)
 
     @app.route('/', methods=['POST'])
+    @login_required
     def create_item():  # pylint:disable=unused-variable
-        Items.add_item(request.form['title'])
+        if app.config['LOGIN_DISABLED'] or current_user.has_write_permission():
+            Items.add_item(request.form['title'])
         return redirect('/')
 
     @app.route('/actions/<action>/<id>')
+    @login_required
     def perform_item_action(action, id):  # pylint:disable=unused-variable
-        if action == 'doing':
-            Items.doing_item(id)
-        elif action == 'done':
-            Items.done_item(id)
+        if app.config['LOGIN_DISABLED'] or current_user.has_write_permission():
+            if action == 'doing':
+                Items.doing_item(id)
+            elif action == 'done':
+                Items.done_item(id)
 
         return redirect('/')
```

Comment:
This is fine, but it's worth considering whether/when it would be nicer for users to see an explicit message if they've tried to do something they don't have permission for - e.g. we could return them an explicit Unauthorized response
For example, if you don't expect users to do this accidentally then doing so might help make it really obvious when something has gone wrong